cpp/src/arrow/compute/kernels/cast.cc

```diff
@@ -1503,19 +1503,18 @@ Status GetCastFunction(const DataType& in_type, std::shared_ptr<DataType> out_ty
     CAST_FUNCTION_CASE(LargeStringType);
     CAST_FUNCTION_CASE(DictionaryType);
     case Type::NA:
-      cast_kernel.reset(new FromNullCastKernel(std::move(out_type)));
```

Comment:
Removed move because of the validation [below](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/cast.cc#L1524) dereferences `out_type`.